docs/calendar-draft.md

```diff
@@ -1,18 +1,50 @@
+This doc describes a design for first-class support for non-Gregorian [calendars](https://en.wikipedia.org/wiki/Calendar) in Temporal.  Although most of this document is based on Temporal.Date, most of this applies to Temporal.DateTime, Temporal.YearMonth, Temporal.MonthDay, and Temporal.Time as well.
 
-## Temporal.Date internal slots
+## Data Model
+
+### Temporal.Date internal slots
+
+Main issue: https://github.com/tc39/proposal-temporal/issues/290
 
 Temporal.Date currently has three internal slots: year, month, and day. (An "internal slot" refers to actual data, as opposed to "properties", which could be computed.)  In this proposal, those slots are renamed to `[[IsoYear]]`, `[[IsoMonth]]`, and `[[IsoDay]]`, and an additional `[[Calendar]]` slot is added.  The calendar slot contains an object implementing the Temporal.Calendar interface, described below.
 
+This data model makes the simple assumption that the concept of a "day" is a solar day (main issues: [#390](https://github.com/tc39/proposal-temporal/issues/390), [#389](https://github.com/tc39/proposal-temporal/issues/389)).  Most or all modern-use calendars, even those with lunar month cycles, use a solar day (based on sunrise) instead of a lunar day (based of moonrise).  For calendars that do use a lunar day, such as the Hawaiian Moon Calendar, a Temporal.DateTime can be used instead of Temporal.Date when the distinction is important.
+
+### Temporal.DateTime and Temporal.Time internal slots
+
+As with Temporal.Date, all of these types will gain a `[[Calendar]]` slot, and year, month, and day will be renamed `[[IsoYear]]`, `[[IsoMonth]]`, and `[[IsoDay]]`.
+
+### Temporal.YearMonth and Temporal.MonthDay internal slots
+
+Main issue: https://github.com/tc39/proposal-temporal/issues/391
+
+For reasons explained above, using the ISO calendar as the internal data model has many advantages.  However, there are several challenges for these two "incomplete" types: lunar months don't line up with solar months, and not every lunar month occurs in every solar year.  After discussing several data model alternatives, we reached the conclusion that the simplest data model for Temporal.YearMonth and Temporal.MonthDay is to make it share the same data model as Temporal.Date, with the following slots:
+
+Temporal.YearMonth:
+
+- `[[IsoYear]]`
+- `[[IsoMonth]]`
+- `[[Calendar]]`
+- `[[RefIsoDay]]`
+
+Temporal.MonthDay:
+
+- `[[IsoMonth]]`
+- `[[IsoDay]]`
+- `[[Calendar]]`
+- `[[RefIsoYear]]`
+
+For calendars that use ISO-style months, such as Gregorian, Solar Buddhist, and Japanese, "RefIsoDay" and "RefIsoYear" can be ignored.  However, for lunar and lunisolar calendars, such as Hebrew, Saudi Arabian Islamic, and Chinese, these additional fields allow those calendars to disambiguate which YearMonth and MonthDay are being represented.
+
 ## Temporal.Calendar interface
```

Comment:
Can we also note here that there's a concrete class, `Temporal.Calendar`, with a constructor that takes a calendar ID (a la 402), and that you can subclass this class (as described in #300)?

Reply:
Does this look correct?